Mathlib/Lean/Expr/Basic.lean

```diff
@@ -148,6 +149,16 @@ def modifyArgM [Monad M] (modifier : Expr → M Expr) (e : Expr) (i : Nat) (n :=
   let a ← modifier a
   return modifyArg (fun _ => a) e i n
 
+/-- Traverses an expression `e` and renames bound variables named `old` to `new`. -/
+def renameBVar (e : Expr) (old new : Name) : Expr :=
+  match e with
+  | app fn arg => app (fn.renameBVar old new) (arg.renameBVar old new)
+  | lam n ty bd bi =>
+    lam (if n == old then new else n) (ty.renameBVar old new) (bd.renameBVar old new) bi
+  | forallE n ty bd bi =>
+    forallE (if n == old then new else n) (ty.renameBVar old new) (bd.renameBVar old new) bi
+  | e => e
```

Comment:
This is a direct port of the Lean 3 code, so it's fine.  But it's got lot of limitations.  For example it doesn't change the bound names in let-expressions (because the .letE case is missing) or in let_fun (because the .mdata case is missing).